src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SlhDsaImplementation.OpenSsl.cs

```diff
@@ -65,6 +64,22 @@ protected override void SignDataCore(ReadOnlySpan<byte> data, ReadOnlySpan<byte>
         protected override bool VerifyDataCore(ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, ReadOnlySpan<byte> signature) =>
             Interop.Crypto.SlhDsaVerifyPure(_key, data, context, signature);
 
+        protected override void SignPreHashCore(ReadOnlySpan<byte> hash, ReadOnlySpan<byte> context, ReadOnlySpan<char> hashAlgorithmOid, Span<byte> destination) =>
+            Helpers.SlhDsaPreHash(
+                hash,
+                context,
+                hashAlgorithmOid,
+                Helpers.CreateStackTuple(_key, destination),
+                static (message, state) => Interop.Crypto.SlhDsaSignPreEncoded(state.Item1, message, state.Item2));
```

Comment:
I'm not a fan of Item1/Item2 appearing in code again, but it's probably not worth the hoops to jump through to avoid it.

As an example, the sign one could call the verify one, downgrading its Span to ReadOnlySpan, then when it gets called back, it can MemoryMarshal convert it back to a writable span before calling its own callback.  But that's... kind of 🤮

Reply:
@krwq also suggested removing the tuples, so I'll get rid of them. Since the destination/signature is an opaque `TState` (the encoder just needs context and message), it should be possible to use `Span` or `ReadOnlySpan` as appropriate without marshalling between them.